etest/build.py

```diff
@@ -0,0 +1,154 @@
+"""Build command for etest Docker images."""
+import os
+from pathlib import Path
+
+import click
+
+from etest import docker
+
+
+@click.command(name="etest-build")
+@click.option("-q", "--quiet", is_flag=True, default=False, help="Suppress all output.")
+@click.option("-v", "--verbose", is_flag=True, default=False, help="Provide more output.")
+@click.option("--hardened/--no-hardened", default=False, help="Use a hardened profile.")
+@click.option("--multilib/--no-multilib", default=True, help="Use a multilib profile.")
+@click.option("--systemd/--no-systemd", default=False, help="Use a systemd profile.")
+# The following could and should be excised to an architecture module that
+# handles the nuances of architecture differences in Gentoo and Docker.
+@click.option(
+    "--architecture",
+    "--arch",
+    type=click.Choice(["amd64", "x86", "arm64", "armv5", "armv7", "ppc64"], case_sensitive=False),
+    default="amd64",
+    help="Architecture for the built image.",
+)
+@click.option(
+    "--libc",
+    type=click.Choice(["glibc", "musl", "uclibc"], case_sensitive=False),
+    default="glibc",
+    help="libc for the built image.",
+)
+@click.option("--path", type=Path, default=os.getcwd(), help="Path to the directory containing the Dockerfile.")
+def main(
+    quiet: bool,
+    verbose: bool,
+    hardened: bool,
+    multilib: bool,
+    systemd: bool,
+    architecture: str,
+    libc: str,
+    path: Path,
+) -> None:
+    """Build the etest image."""
+    img_profile = build_profile(quiet, architecture, libc, hardened, multilib, systemd)
```

Comment:
Hmm this suggests we should have an etest/build/profile.py module with this concept encapsulated.

Reply:
True. We will probably also need it for the base etest command too.

Reply:
Yep, it's a pretty generic concept for this project it looks like.